dependencies.yaml

```diff
@@ -441,3 +444,25 @@ dependencies:
           - pandas
           - *scikit_learn
           - seaborn
+  depends_on_distributed_ucxx:
+    common:
+      - output_types: conda
+        packages:
+          # UCXX is not currently a hard-dependency thus only installed during tests,
+          # this will change in the future.
```

Comment:
My intent was to make `ucxx` an optional dependency of `raft-dask` although I now realize that's not the case in https://github.com/rapidsai/raft/blob/branch-24.06/conda/recipes/raft-dask/meta.yaml . I think it's ok to have `ucxx` (and thus `libucxx`) pulled in by RAFT -- even though it's technically not a hard dependency because it won't be activated by default -- but it will bloat a bit the installation. What I'm saying is we have two options:
1. Make `ucxx` a soft dependency of `raft-dask` and then leave this here; or
2. Accept `ucxx` as a hard dependency of `raft-dask` and remove this.

Do you have thoughts?

Reply:
Let's just stick with a hard dependency for now. It's easier to manage and that is the long-term plan anyway.

Reply:
I was just thinking a bit more about this and perhaps it should indeed be an optional dependency everywhere, both in `raft-dask` and `cuml`. Since UCX is not a requirement to run either one (people may choose NCCL as a replacement, for example) it isn't the case that users always require either UCX-Py (or UCXX).

If it makes things simpler for us for the moment, I think it's ok to make it a hard dependency, but in the long run I don't think it makes sense from the packaging perspective to require UCX-Py/UCXX. WDYT @vyasr ?

Reply:
Ultimately raft-dask is going to have to be compiled against libucxx, right? The C++ dependency will be there no matter what, unless you plan to rewrite the raft C++ to use a dlopen of ucxx.